prometheus_client/openmetrics/parser.py

```diff
@@ -307,12 +307,17 @@ def _parse_nh_sample(text, suffixes):
 
 def _parse_nh_struct(text):
     pattern = r'(\w+):\s*([^,}]+)'
-    
-    re_spans = re.compile(r'(positive_spans|negative_spans):\[(\d+:\d+,\d+:\d+)\]')
+    re_spans = re.compile(r'(positive_spans|negative_spans):\[(\d+:\d+(,\d+:\d+)*)\]')
     re_deltas = re.compile(r'(positive_deltas|negative_deltas):\[(-?\d+(?:,-?\d+)*)\]')
```

Comment:
That middle question mark looks off to me

```suggestion
    re_deltas = re.compile(r'(positive_deltas|negative_deltas):\[(-?\d+(:,-?\d+)*)\]')
```

Reply:
I’m afraid it cannot be omitted. My understanding is that this `(?:,-?\d+)*` matches a pattern like this `1,2,-3`, whereas `(:,-?\d+)*` would end up matching something like this `1:,2:,-3:`. In any case, if I remove the question mark there, the related tests fail :/

Reply:
Ah, I didn't know that `(?:<exp>)` was special syntax. https://docs.python.org/3/library/re.html#regular-expression-syntax . Ack.